src/window.rs

```diff
@@ -28,16 +29,13 @@ impl Window {
             .create_surface(drm_display.width, drm_display.height, format, usage)
             .unwrap();
 
-        Window { gbm_surface, drm_display, crtc_set: false }
+        Window { gbm_surface, drm_display, frame_count: 0, previous_bo: None }
     }
 
-    // TODO(mbernat): Add a "Frame" abstraction that calls `swap_buffers` internally when
-    // it's finished (just like glium's Frame does) so that users don't need to bother with this.
-
     /// # Safety
     /// this must be called exactly once after `eglSwapBuffers`,
     /// which happens e.g. in `Frame::finish()`.
-    pub unsafe fn swap_buffers(&mut self) {
+    unsafe fn present(&mut self) {
```

Comment:
Does this still need to be marked unsafe?

Reply:
I think so, we cannot guarantee that the surface's buffers have been used properly from inside the function.

But it would be great if we could understand the safety situation better and make the preconditions clearer.